zookeeper-server/src/main/java/org/apache/zookeeper/backup/NFSBackupStorage.java

```diff
@@ -0,0 +1,324 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Implementation for NFS-based backup storage provider
+ */
+public class NFSBackupStorage implements BackupStorageProvider {
+  private static final Logger LOG = LoggerFactory.getLogger(NFSBackupStorage.class);
+
+  public static final String TMP_FILE_PREFIX = "TMP_";
+  private static final String PATH_SEPARATOR = File.separator;
+  private final BackupConfig _backupConfig;
+  private final String _fileRootPath;
+
+  /**
+   * Constructor using BackupConfig to get backup storage info
+   * @param backupConfig The information and settings about backup storage, to be set as a part of ZooKeeper server config
+   */
+  public NFSBackupStorage(BackupConfig backupConfig) {
+    _backupConfig = backupConfig;
+    _fileRootPath =
+        String.join(PATH_SEPARATOR, _backupConfig.getMountPath(), _backupConfig.getNamespace());
+  }
+
+  @Override
+  public BackupFileInfo getBackupFileInfo(File file) throws IOException {
+    String backupFilePath = constructBackupFilePath(file.getName());
+    File backupFile = new File(backupFilePath);
+
+    if (!backupFile.exists()) {
+      throw new BackupException(
+          "Backup file " + file.getName() + " does not exist, could not get file info.");
+    }
+
+    BasicFileAttributes fileAttributes =
+        Files.readAttributes(Paths.get(backupFilePath), BasicFileAttributes.class);
+    return new BackupFileInfo(backupFile, fileAttributes.lastModifiedTime().toMillis(),
+        fileAttributes.size());
+  }
+
+  @Override
+  public List<BackupFileInfo> getBackupFileInfos(File path, String prefix) throws IOException {
+    String backupDirPath = constructBackupFilePath(path.getPath());
+    File backupDir = new File(backupDirPath);
+
+    if (!backupDir.exists()) {
+      throw new BackupException(
+          "Backup directory " + path.getPath() + " does not exist, could not get file info.");
+    }
+
+    // Get a list of files whose file names start with the provided prefix
+    FilenameFilter fileFilter = (dir, name) -> name.startsWith(prefix);
+    File[] files = backupDir.listFiles(fileFilter);
+    if (files == null) {
+      throw new BackupException("The provided directory path " + path.getPath()
+          + " is invalid, could not get file info.");
+    }
+
+    // Read the file info and add to the list. If an exception is thrown, the entire operation will fail
+    List<BackupFileInfo> backupFileInfos = new ArrayList<>();
+    for (File file : files) {
+      BasicFileAttributes fileAttributes =
+          Files.readAttributes(Paths.get(file.getPath()), BasicFileAttributes.class);
+      backupFileInfos.add(new BackupFileInfo(file, fileAttributes.lastModifiedTime().toMillis(),
+          fileAttributes.size()));
+    }
```

Comment:
This is duplicate functions with getBackupFileInfo. Why dont just directly call getBackupFileInfo?